code/game/objects/structures/mirror.dm

```diff
@@ -23,32 +36,39 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
 	if(broken || !Adjacent(user))
 		return
 
-	if(ishuman(user))
-		var/mob/living/carbon/human/H = user
-
-		//see code/modules/mob/dead/new_player/preferences.dm at approx line 545 for comments!
-		//this is largely copypasted from there.
-
-		//handle facial hair (if necessary)
-		if(H.gender != FEMALE)
-			var/new_style = input(user, "Select a facial hairstyle", "Grooming")  as null|anything in GLOB.facial_hairstyles_list
-			if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
-				return //no tele-grooming
-			if(new_style)
-				H.facial_hairstyle = new_style
-		else
-			H.facial_hairstyle = "Shaved"
-
-		//handle normal hair
-		var/new_style = input(user, "Select a hairstyle", "Grooming")  as null|anything in GLOB.hairstyles_list
-		if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+	if(!ishuman(user))
+		return
+
+	mirror_stuff(user)
+
+//Mirror man, mirror man, does whatever a mirror can.
+/obj/structure/mirror/mirror_stuff(mob/living/carbon/human/stylist)
+	//see code/modules/mob/dead/new_player/preferences.dm at approx line 545 for comments!
+	//this is largely copypasted from there.
+
+	//handle normal hair
+	if(!HAS_TRAIT(stylist, TRAIT_BALD))
+		var/new_style = input(stylist, "Select a hairstyle", "Grooming")  as null|anything in GLOB.hairstyles_list
```

Comment:
`tgui_input_list()` is the tgui version of input, i'm gonna say it's a better idea to use that here